packages/pilot/src/containers/CreateTransaction/Products/Shipping.js

```diff
@@ -0,0 +1,234 @@
+import React, { Fragment } from 'react'
+import PropTypes from 'prop-types'
+import {
+  complement,
+  isEmpty,
+  when,
+} from 'ramda'
+import {
+  CardContent,
+  CardTitle,
+  Col,
+  FormDropdown,
+  FormInput,
+  Row,
+} from 'former-kit'
+import Form from 'react-vanilla-form'
+import CurrencyInput from '../../../components/CurrencyInput'
+import style from './style.css'
+
+import isRequired from '../../../validation/required'
+import isValidDate from '../../../validation/date'
+import lt from '../../../validation/lessThan'
+
+import statesModel from '../../../models/states'
+import countriesModel from '../../../models/countries'
+
+const stateOptions = statesModel.map(state => ({
+  name: state.code,
+  value: state.code,
+}))
+
+const countriesOptions = countriesModel.map(country => ({
+  name: country.name,
+  value: country.code,
+}))
+
+const validations = t => ({
+  address: {
+    city: isRequired(t('required')),
+    country: isRequired(t('required')),
+    state: isRequired(t('required')),
+    street: isRequired(t('required')),
+    streetNumber: isRequired(t('required')),
+    zipcode: isRequired(t('required')),
+  },
+  deliveryDate: when(
+    complement(isEmpty),
+    isValidDate(t('validations.isDate'))
+  ),
+  fee: [isRequired(t('required')), lt(0, t('positive_value'))],
+  name: isRequired(t('required')),
+})
+
+const ShippingForm = ({
+  data,
+  onChange,
+  onChangeWithMask,
+  t,
+}) => (
+  <Fragment>
+    <CardTitle title={t('add_transaction_shipping_title')} />
+
+    <CardContent>
+      <Form
+        data={data}
+        onChange={onChange}
+        validation={validations(t)}
+        validateOn="blur"
+      >
+        <fieldset className={style.fieldset} name="address">
+          <Row>
+            <Col desk={2}>
+              <FormInput
+                name="zipcode"
+                label={t('add_transaction_shipping_zipcode')}
+                mask="11111-111"
+                onChange={onChangeWithMask('address.zipcode')}
```

Comment:
Adoro _function currying_ mas aqui me parece meio estranho não ? 🤔 
Pq a função `onChangeWithMask` retorna uma função que vai ser o `onChange` em sí, será que não vale um rename ?
( Exemplo bem tosco )
```suggestion
  onChange={getOnChangeWithMaskFunction('address.zipcode')}
```

Reply:
@Deivis no momento eu deixei como estava, porém vou abrir um PR no former-kit pra podermos alterar essa lib de mask, assim evitamos usar essas funções pra atualizar os dados no state.